server/graphql/document/src/types/program.rs

```diff
@@ -0,0 +1,42 @@
+use async_graphql::*;
+use graphql_core::ContextExt;
+
+use super::{document::DocumentNode, encounter::EncounterNode};
+
+pub struct ProgramNode {
+    pub store_id: String,
+    pub patient_id: String,
+    pub program: String,
+    pub document_node: DocumentNode,
+}
+
+#[Object]
+impl ProgramNode {
+    /// The encounter document
+    pub async fn document(&self) -> &DocumentNode {
+        &self.document_node
+    }
+
+    /// The program document
+    pub async fn encounters(&self, ctx: &Context<'_>) -> Result<Vec<EncounterNode>> {
+        // TODO use loader
+        let context = ctx.service_provider().context()?;
+        let entries = ctx
+            .service_provider()
+            .patient_service
+            .get_patient_program_encounters(
```

Comment:
If there are many encounters and many edits on those encounters, we would want to select the latest edit for each encounters (not all of the edits and all of the counters), is that already implemented ?

Reply:
You only get the latest encounter. But currently there is no pagination to limit how many encounters you get back. For example, if you have 1000 encounters you always get the latest version of these 1000 document.